auth.go

```diff
@@ -257,6 +258,26 @@ func (s *Service) AddDevProvider(port int) {
 	s.providers = append(s.providers, provider.NewService(provider.NewDev(p)))
 }
 
+// AddAppleProvider allow SignIn with Apple ID
+func (s *Service) AddAppleProvider(name string, appleConfig provider.AppleConfig, privKeyLoader provider.PrivateKeyLoaderInterface) error {
+	p := provider.Params{
+		URL:        s.opts.URL,
+		JwtService: s.jwtService,
+		Issuer:     s.issuer,
+		//AvatarSaver: s.avatarProxy, // not implemented avatar for AppleProvider yet
+		L: s.logger,
+	}
+
+	// Error checking at create need for catch one when apple private key init
+	appleProvider, err := provider.NewApple(name, p, appleConfig, oauth2.Endpoint{}, privKeyLoader)
+	if err != nil {
+		return err
```

Comment:
I propose to wrap it into something like "problem creating an AppleProvider"

Reply:
added error wrap in [8f262ce](https://github.com/go-pkgz/auth/pull/86/commits/8f262cef53e491146a7af03b723e7df6972c7eb7) commit